docs/man/gpio.txt

```diff
@@ -0,0 +1,118 @@
+GENERIC GPIO(8)
+=================
+
+NAME
+----
+
+GENERIC GPIO - Driver for GPIO connected UPS
+
+SYNOPSIS
+--------
+
+*GENERIC_GPIO* -h
+
+*GENERIC_GPIO* -a 'GPIOCHIP0' ['OPTIONS']
+
+NOTE: This man page only documents the specific features of the *GPIO*
+driver. For information about the core driver, see linkman:nutupsdrv[8].
+
+SUPPORTED HARDWARE
+------------------
+
+This is the driver for GPIO attached UPS devices.
+
+The driver has been tested against CyberPower CyberShield CSN27U12V attached to
+Orange Pi Zero GPIO.
+
+More information about this UPS can be found here: ::
+https://www.cyberpowersystems.com/resources/csn27u12v-um/
+
+
+EXTRA ARGUMENTS
+---------------
+
+This driver supports the following optional settings in the
+linkman:ups.conf[5] file:
+
+Driver control:
+~~~~~~~~~~~~~~
+
+*rules*='value'::
+A string consisting of sub-strings. Each sub-string describes GPIO line
+states conversion formula to specific NUT state, like
+nut_state=[^]line_num[logical_operation.line_num]. Not (^) , and (&) , or
+(|)operations are supported for now. nut_state should correspond to NUT
+state, line_num to GPIO line number connected to UPS open collector pin.
+CyberShield CSN27U12V describes pins as:
+ON BATTERY			Low when operating from utility line
```

Comment:
I think asciidoc table markup (with pipe characters) would do well here, and this may need blank lines around to differentiate from other paragraphs. If possible, build with HTML docs enabled to iterate the result-oriented markup more easily (or use an IDE with an asciidoc plugin to render it real-time as you type) :)